File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i,j,res = 0;
        for (i = 0,j = 1; j < prices.size(); j++){
            if (prices[i] < prices[j]){
                if (res < prices[j] - prices[i])
                    res = prices[j] - prices[i];
            }
            else
                i=j;
        }
        if (i == prices.size()-1 && res == 0)
            return 0;
        return res;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy_price = prices[0];

        int max_profit = 0;

        for (int i = 1; i < prices.length; i++){
            
            if (prices[i] < buy_price){
                buy_price = prices[i];
            }
            else{
                int curr_profit = prices[i] - buy_price;
                max_profit = Math.max(curr_profit,max_profit);
            }
        }
        return max_profit;
    }
}